probeye/inference/torch/solver.py

```diff
@@ -395,6 +395,14 @@ def loglike(self, theta: th.Tensor):
             A vector of pyro.samples (i.e. tensors) for which the log-likelihood
             function should be evaluated.
         """
+        
+        # check whether the values of the latent parameters are within the specified domain.
+        # if they are outside of the domain, break calculating the log-likelihood function
+        # and return instead of it minus infinity.
+        
+        if self.problem.check_parameters_domain(theta) == False:
+            return -np.inf
```

Comment:
Same thing as before.
```suggestion
        if not self.problem.check_parameters_domain(theta):
            return -np.inf
```